### ISSUE TITLE
Android previewer gives exception for F# proejcts

### ISSUE THREAD

**Issue by dsyme** [mono/monodevelop#4062]
Repro on VS4Mac master (when https://github.com/xamarin/md-addins/pull/2956 is included)
* VSforMac
* File --> New Project --> Blank Forms App --> F# 
* Click on the AppPage.xaml, Android previewer:

This exception shows in the previewer:
```
System.Exception: Project not part of workspace
  at MonoDevelop.Ide.TypeSystem.TypeSystemService+<GetCodeAnalysisProjectAsync>d__79.MoveNext () [0x000d1] in /Users/builder/data/lanes/432/9326b144/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs:266 
--- End of stack trace from previous location where exception was thrown ---
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner.CodeInteractions/CodeModelBridge.cs:44 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner.CodeInteractions/CodeModelBridge.cs:300 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner/DesignerProject.cs:300 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner/DesignerProject.cs:307 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner/DesignerProject.cs:143 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner/AndroidDesignerService.cs:464 
...
<InitializeAndroidDesignerProject>d__65.MoveNext () [0x00122] in /Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Forms/Xamarin.FormsPreviewer.MonoDevelop/Xamarin.Designer.Forms/XamarinStudioXamlPreviewerView.cs:535 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Forms/Xamarin.Designer.Forms.Preview/Android/AndroidXamlRenderer.cs:78 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Forms/Xamarin.Designer.Forms.Preview/Android/AndroidXamlRenderer.cs:122 
...
/Users/vsts/agent/2.129.1/work/1/s/Xamarin.Designer.Forms/Xamarin.Designer.Forms.Preview/Android/AndroidXamlRenderer.cs:197 ...
```

> VS bug #579690

**Comment by dsyme** [comment]
Exception is being thrown here: https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs#L266

And here: https://github.com/xamarin/designer/blob/master/Xamarin.Designer.Android/Xamarin.AndroidDesigner/Xamarin.AndroidDesigner.CodeInteractions/CodeModelBridge.cs#L44

It seems as though the Android designer is actually prepared to cope with no Roslyn information being available. 

My comment on the PR that caused this:
> It looks like [this change](https://github.com/mono/monodevelop/commit/6372601cf428eb01905f59b4c2f99aa70c3ba1d5#diff-0524bda4b72240c35996595524f79113R260) which is part of this PR means that an exception is now being raised when Roslyn Workspaces project information is not available, e.g. for F# projects, see https://github.com/mono/monodevelop/issues/4062.
>
> The Android designer depends on null being raised for this condition, and is prepared to handle it.
> 
> Is it ok if we remove the throw of the exception and return null in this method as before?  The corresponding `GetCodeAnalysisProject` just before `GetCodeAnalysisProjectAsync` does return null in this case.

**Comment by nosami** [comment]
Fixed in master and 15.7

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4109 merged by: Jason Imison  
 Author: mono  
 Commit: bf18e9c3e0fdead53fb26b97f974d06ef3dab811 (xamarin/monodevelop)
